### PR TITLE
RM-1492 BocList: change cell content wrappers from span to div

### PR DESCRIPTION
**Pull request by Frazerus**
https://re-motion.atlassian.net/browse/RM-1492

All tests are green, the few that contain spans are about spans where data is displayed, which therefore were not changed to divs.

Already tested successfully that this change also fixes the problem fixed in https://re-motion.atlassian.net/browse/RM-8656. 

PS: Still a draft until the teamcity builds finish for all webtests.

### CODE REVIEW

**Review by a-ctor — COMMENTED** [review]
request me again

**Review by a-ctor — COMMENTED** [review]
The changes look okay but I don't have the knowledge to do an actual review so I will hand it over to @MichaelKetting  instead.

File: Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocListImplementation/Rendering/BocCompoundColumnRendererTest.cs [hunk 1]
```diff
@@ -70,7 +70,7 @@ public void RenderEmptyCell ()
       Html.AssertAttribute(td, "class", _bocListCssClassDefinition.DataCell);
       Html.AssertAttribute(td, "role", "cell");
 
-      var span = Html.GetAssertedChildElement(td, "span", 0);
+      var span = Html.GetAssertedChildElement(td, "div", 0);
```
Comment (MichaelKetting) [inline Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocListImplementation/Rendering/BocCompoundColumnRendererTest.cs:73]: Those variables should also get a rename, please.

File: Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocListImplementation/Rendering/BocCompoundColumnRendererTest.cs [hunk 2]
```diff
@@ -176,7 +176,7 @@ public void RenderEnforcedWidthCell ()
       Html.AssertStyleAttribute(cropSpan, "overflow", "hidden");
       Html.AssertStyleAttribute(cropSpan, "white-space", "nowrap");
 
-      var span = Html.GetAssertedChildElement(cropSpan, "span", 0);
```
Comment (MichaelKetting) [inline Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocListImplementation/Rendering/BocCompoundColumnRendererTest.cs:179]: Here is also a parent element `cropSpan` referenced, which indicates we now try to nest a `div` inside a `span` which is considered a big no-no. Checking further up in the test, `cropSpan` is indeed a `span` so there is a change still missing.

**Review by MichaelKetting — CHANGES_REQUESTED** [review]
CSS is not yet checked. 

I did find a missing span/div conversion.

I checked the BocList-control objects in web testing and found a couple of `span` usages. It suprises me that the tests didn't fail for those. Please verify that those `span` are indeed correct and optimally, check why the test doesn't fail.

`BocListControlObject.GetCellWhereColumnContainsExactly`
`BocListControlObject.GetCellWhereColumnContains`

cc @a-ctor

**Comment by Frazerus** [comment]
`BocListControlObject.GetCellWhereColumnContainsExactly`
`BocListControlObject.GetCellWhereColumnContains`
dont fail because the bocListCommandText, which houses the visible text that the method searches for, is still a span. 

As it does not have any child elements and only text inside, it should probably stay as a span.

File: Remotion/ObjectBinding/Web/Res/Themes/NovaViso/HTML/BocList.css [hunk 3]
```diff
@@ -127,7 +127,6 @@ div.bocList table.bocListTable td.bocListTitleCell > span
 }
 
 div.bocList table.bocListTable td.bocListDataCell > a:not([hidden]),
-div.bocList table.bocListTable td.bocListDataCell > span:not([hidden]),
 div.bocList table.bocListTable td.bocListDataCell > div:not([hidden])
```
Comment (MichaelKetting) [inline Remotion/ObjectBinding/Web/Res/Themes/NovaViso/HTML/BocList.css:130]: :D And you wonder why that was never a red flag :D

File: Remotion/ObjectBinding/Web/Res/Themes/NovaViso/HTML/BocList.css [hunk 4]
```diff
@@ -183,8 +180,8 @@ div.bocList table.bocListTable td.bocListDataCell *.bocListCellStructureElement
 div.bocList table.bocListTable th.bocListTitleCell > a,
 div.bocList table.bocListTable td.bocListDataCell > a,
 div.bocList table.bocListTable td.bocListDataCell *.bocListCellStructureElement > a,
-div.bocList table.bocListTable td.bocListDataCell > span:not(.bocListCellStructureElement),
-div.bocList table.bocListTable td.bocListDataCell > span:not(.bocListCellStructureElement),
+div.bocList table.bocListTable td.bocListDataCell > div:not(.bocListCellStructureElement),
+div.bocList table.bocListTable td.bocListDataCell > div:not(.bocListCellStructureElement),
```
Comment (MichaelKetting) [inline Remotion/ObjectBinding/Web/Res/Themes/NovaViso/HTML/BocList.css:184]:
```suggestion
div.bocList table.bocListTable td.bocListDataCell > div:not(.bocListCellStructureElement),
```
I think the original code has a copy&paste duplicate here.

File: Remotion/ObjectBinding/Web/Res/Themes/NovaViso/HTML/BocList.css [hunk 5]
```diff
@@ -183,8 +180,8 @@ div.bocList table.bocListTable td.bocListDataCell *.bocListCellStructureElement
...
 div.bocList table.bocListTable td.bocListDataCell *.bocListCellStructureElement > div:not(.bocListCellStructureElement),
 div.bocList table.bocListTable td.bocListDataCell *.bocListCellStructureElement > div:not(.bocListCellStructureElement)
```
Comment (MichaelKetting) [inline Remotion/ObjectBinding/Web/Res/Themes/NovaViso/HTML/BocList.css:186]:
```suggestion
div.bocList table.bocListTable td.bocListDataCell *.bocListCellStructureElement > div:not(.bocListCellStructureElement)
```
Copy&Paste duplicate, unless I'm blind.

**Review by MichaelKetting — COMMENTED** [review]
now for CSS testing...

File: Remotion/ObjectBinding/Web/Res/Themes/NovaViso/HTML/BocList.css (hunk 3 above)
Comment (Frazerus) [reply Remotion/ObjectBinding/Web/Res/Themes/NovaViso/HTML/BocList.css:130]: I dont think I understand ;/

**Review by MichaelKetting — COMMENTED** [review]
now there's just my css review left

**Comment by MichaelKetting** [comment]
Reminder: Breaking Change ticket for updated markup (one ticket, lists which spans have been switched, e.g `span.bocListContent`, ...)

**Review by MichaelKetting — APPROVED** [review]
@Frazerus I thinkI found the remaining issues / acceptable changes and added a commit for this. Please review and integrate :)

**Pull request merged**